gomock/matchers.go

```diff
@@ -170,23 +170,30 @@ func (n notMatcher) String() string {
 }
 
 type regexMatcher struct {
-	regex string
+	regex      *regexp.Regexp
+	compileErr error
 }
 
 func (m regexMatcher) Matches(x any) bool {
-	str, ok := x.(string)
-	if !ok {
+	if m.regex == nil {
 		return false
 	}
-	match, err := regexp.MatchString(m.regex, str)
-	if !match || err != nil {
+
+	switch t := x.(type) {
+	case string:
+		return m.regex.MatchString(t)
+	case []byte:
+		return m.regex.Match(t)
+	default:
 		return false
 	}
-	return true
 }
 
 func (m regexMatcher) String() string {
-	return "matching regex " + m.regex
+	if m.compileErr != nil {
+		return m.compileErr.Error()
+	}
+	return "matching regex " + m.regex.String()
```

Comment:
```suggestion
	return "matches regex " + m.regex.String()
```
Reason: this library has been using something like "has ...", "is ...", or symbols. "matches" seems to match the existing style better.